changelog.d/260.deprecation.rst

```diff
@@ -0,0 +1,8 @@
+Added new option to ``@attr.s()``: *replace_class* will make ``attrs`` stop attaching methods to the existing class.
+Instead it will create a new class with the methods attached.
+This is consistent with how ``slots``-classes have *always* been created.
+
+For the next year, the default value is ``replace_classes=False`` which is the current behavior.
+
+The default value will change to ``True`` no sooner than one year after this release.
+There are no concrete plans to remove ``replace_classes=False`` altogether but its use is discouraged.
```

Comment:
After a year, if no problems present themselves, I say we remove the option altogether. Mainly in the interest of simplicity, `attr.s` seems to be getting options all the time.

Reply:
The problem is that we’d break everyone’s software who decided to do the right thing and set it to true.

Reply:
I guess what we could do is to add a `**kw` and spit out DeprecationWarnings on certain arguments that have been removed.  But I really don’t want `attrs` to become notorious for breaking backward-comparability all the time.  We’re already getting comments like https://www.reddit.com/r/Python/comments/74fro5/attrs_iii_frozen_classes/dnzjn6d/ :|

Reply:
Why don't we just do this: default to False for one year, default to True for one year, remove if no problems. I mean, the alternative is to carry this forever :/

I'm ok either way btw, this is definitely not a hill I want to die on :) I'm just worried about attr.s having a gazillion args down the line, which will make it harder to use.

Reply:
Yeah well I’m deeply unhappy with every option, hence trying to source opinions. :)  The two years plan is something I considered too.  Let's see we can get some opinions out of glyph and Nate.